hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -380,4 +422,17 @@ private object HoodieMergeOnReadRDD {
       case (nullable, _) => nullable
     }
   }
+
+  trait AvroDeserializerSupport extends SparkAdapterSupport {
```

Comment:
this looks like a generic trait.. maybe place in a more common class?

Reply:
Nah, not really -- it requires you to provide required schema, so pretty much falls back into the same bucket as `createAvroSerializer` (in my comment above).

The only reason i've extracted it to a separate Trait is to trim down Log Iterator impl, moving out all the stuff which isn't really tied to iteration.